### ISSUE TITLE
Colorfill Plot throws Exception changing to log scale

### ISSUE THREAD

**Issue by DavidFair** [mantidproject/mantid#28226]
### Expected behavior
- Should be able to change the colour scale to log
### Actual behavior
- Throws exception
```
  File "/opt/mantidnightly/lib/workbench/plotting/figureinteraction.py", line 743, in _change_colorbar_axes
    datafunctions.update_colorbar_scale(self.canvas.figure, image, scale_type, image.norm.vmin, image.norm.vmax)
  File "/opt/mantidnightly/lib/mantid/plots/datafunctions.py", line 1024, in update_colorbar_scale
    if locator.tick_values(vmin=vmin, vmax=vmax).size == 0:
  File "/usr/lib/python3/dist-packages/matplotlib/ticker.py", line 2123, in tick_values
    "Data has no positive values, and therefore can not be "
ValueError: Data has no positive values, and therefore can not be log-scaled.
```

### Steps to reproduce the behavior
Run the following:
```
Load(Filename='GEM38370_Focussed.nxs', OutputWorkspace='GEM38370_Focussed')
ConvertUnits(InputWorkspace='GEM38370_Focussed', OutputWorkspace='ws_lambda', Target='Wavelength')
```
- Plot -> Colorfill
- Right click figure -> Colour Bar -> Log -> Stops working

**Comment by PhilColebrooke** [comment]
So this issue happens when you try to change the colorbar scale to log and the scale's range contains negative values.

There's already a check for if the min value is zero so the simple way to fix it is to also check if the min value is less than zero, and adjust it accordingly.

However when looking into this I found that the Images tab in the figure options restricts the input of the min and max value so you can't make them less than zero, which raises a question of whether negative values should be allowed on the colorbar scale. Either way what's allowed in the Images tab and what's allowed in the dialog when you double-click the colorbar axis should match.

Any ideas @DanielMurphy22?

**Comment by DavidFair** [comment]
@PhilColebrooke Out of curiosity how did Plot handle this?

**Comment by PhilColebrooke** [comment]
I forgot Plot existed and didn't think to check what it does 😅 

When the scale is set to linear, any range is permitted.
When the scale is set to log and the min value is zero or negative, it is changed to slightly-above-zero.
When the scale is set to log and both the min and max values are negative, the range is changed to 1 to 1000 (?)

I think making Workbench match this behaviour is too big a task for this release since it's dealing with two different menus. Should a temporary fix be put in place in the meantime or should fixing this issue be postponed?

**Comment by Harrietbrown** [comment]
Fixed by #28242
